magento/magento2
# Fix to take the real customer store id when subscriber is created fro…

…m API





### Description (*)
When subscribers are created via customer API call, they are saved with always with the default store_id, eaven if in the customer API call' data the store_id is specified.
The fix uses the real customer store_id.
No need to load it with the storeManager, the information is passed by the customer.

### Related Pull Requests


### Fixed Issues (if relevant)



### Manual testing scenarios (*)

1. Setup magento as multi website and multi store
2. Create a customer posting on /rest/V1/customers endpoint specifying a store_id different from default and inserting into the extension attributes is_subscribed as true
3. Check the newsletter subscriber table. You should found a subscriber saved with the default store_id instead of the store_id you specified in you API call
4. Apply the fix
5. Create a new customer via API
6. Check into the newsletter subscriber table that the new subscriber has the same store_id of its related customer

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#34436: Fix to take the real customer store id when subscriber is created fro…
